### PR TITLE
add new api testcases

### PR DESCRIPTION
**Pull request by zhangliuyan66**

### CODE REVIEW

**Comment by StorageGhoul** [comment]
Can one of the admins verify this patch?

**Review by vojtechtrefny — CHANGES_REQUESTED** [review]
Thank you, looks good to me in general, but there are some issues that should be fixed.

File: tests/crypto_test.py [hunk 1]
```diff
@@ -266,6 +266,12 @@ def test_luks_format_key_size(self):
         key_size = self._get_luks1_key_size(self.loop_dev)
         self.assertEqual(key_size, 256)
 
+    @tag_test(TestTags.SLOW, TestTags.CORE)
+    def test_luks_format_luks2_blob(self):
```
[the remark this answers is not in the document]
Comment (zhangliuyan66) [reply tests/crypto_test.py]: ok，It has been modified

File: tests/crypto_test.py [hunk 2]
```diff
@@ -327,6 +333,22 @@ def test_luks2_resize(self):
         succ = BlockDev.crypto_luks_close("libblockdevTestLUKS")
         self.assertTrue(succ)
 
+    @tag_test(TestTags.SLOW)
+    def test_luks_resize_lucks2_blob(self):
```
[the remark this answers is not in the document]
Comment (zhangliuyan66) [reply tests/crypto_test.py]: ok，It has been modified

File: tests/mpath_test.py [hunk 3]
```diff
@@ -49,6 +49,16 @@ def test_is_mpath_member(self):
...
         self.assertFalse(BlockDev.mpath_is_mpath_member("/dev/loop0"))
 
+    def test_get_mpath_members(self):
+        """Verify that get_mpath_members works as expected"""
+        os.system('service multipathd start')
+        ret = BlockDev.mpath_get_mpath_members()
+        self.assertIsNotNone(ret)
+    def test_set_friendly_names(self):
+        """Verify that set_friendly_names works as expected"""
+        succ = BlockDev.mpath_set_friendly_names(True)
```
[the remark this answers is not in the document]
Comment (zhangliuyan66) [reply tests/mpath_test.py:61]: It has been  add tag，Can you help see if it works

File: tests/mpath_test.py [hunk 4]
```diff
@@ -49,6 +49,16 @@ def test_is_mpath_member(self):
         # device and no error is reported
         self.assertFalse(BlockDev.mpath_is_mpath_member("/dev/loop0"))
 
+    def test_get_mpath_members(self):
+        """Verify that get_mpath_members works as expected"""
+        os.system('service multipathd start')
```
[the remark this answers is not in the document]
Comment (zhangliuyan66) [reply tests/mpath_test.py]: At the beginning, an error was reported, so this operation was added.Closing it later works fine.

File: tests/mpath_test.py [hunk 5]
```diff
@@ -49,6 +49,22 @@ def test_is_mpath_member(self):
...
+    @tag_test(TestTags.NOSTORAGE)
+    def test_get_mpath_members(self):
+        """Verify that get_mpath_members works as expected"""
+        ret = BlockDev.mpath_get_mpath_members()
+        self.assertIsNotNone(ret)
+
+    @tag_test(TestTags.NOSTORAGE)
+    def test_set_friendly_names(self):
+        """Verify that set_friendly_names works as expected"""
+        flag=os.system('ls /etc/multipath.conf')
```
Comment (vojtechtrefny) [inline tests/mpath_test.py:61]: Please use `shutil.which` here to check for `mpathconf`. If you need to check for the file existence then use `os.path.exists`. Thanks.

File: tests/crypto_test.py [hunk 6]
```diff
@@ -148,6 +148,10 @@ def test_luks2_format(self):
         # the simple case with password blob
         succ = BlockDev.crypto_luks_format_blob(self.loop_dev, "aes-xts-plain64", 0, [ord(c) for c in PASSWD], 0)
         self.assertTrue(succ)
+        
+        #test luks2_blob
```
Comment (vojtechtrefny) [inline tests/crypto_test.py:152]: You can remove the `BlockDev.crypto_luks_format_blob` call above.

**Review by vojtechtrefny — CHANGES_REQUESTED** [review]
Two more nitpicks, but other than that this should be good to go. Thanks.

File: tests/crypto_test.py (hunk 6 above)
Comment (zhangliuyan66) [reply tests/crypto_test.py]: ok

File: tests/mpath_test.py (hunk 5 above)
Comment (zhangliuyan66) [reply tests/mpath_test.py]: ok，It has been modified

File: tests/crypto_test.py (hunk 6 above)
Comment (zhangliuyan66) [reply tests/crypto_test.py]: It has been removed，thanks for your suggestion

**Review by vojtechtrefny — APPROVED** [review]
Thank you, looks good to me now.

**Pull request merged**